src/platform/packages/private/kbn-reporting/get_csv_panel_actions/panel_actions/get_csv_panel_action.tsx

```diff
@@ -164,7 +189,8 @@ export class ReportingCsvPanelAction implements ActionDefinition<EmbeddableApiCo
    * Requires `xpack.reporting.csv.enablePanelActionDownload: true` in kibana.yml
    * @deprecated
    */
-  private executeDownload = async (params: ExecutionParams) => {
+  private executeDownload = async (params: ExecutionParamsOld) => {
+    // Deprecated and does not support ES|QL mode
```

Comment:
The conflict was in how to handle the Immediate Download action. Since ES|QL is not supported for this action, I left it as is. The default behaviour without `xpack.reporting.csv.enablePanelActionDownload: true` is fixed with this backport.